### PR TITLE
[EspaceCandidate] Add delegation

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

**Comment by illusionOfParadise** [comment]
[ch12626]

File: src/Entity/Adherent.php [hunk 1]
```diff
@@ -2453,6 +2455,50 @@ public function isCandidate(): bool
...
     }
 
+    public function isDelegatedCandidate(): bool
+    {
+        return \count($this->getReceivedDelegatedAccessOfType('candidate')) > 0;
+    }
+
+    public function isDelegatedHeadedRegionalCandidate(): bool
+    {
+        foreach ($this->receivedDelegatedAccesses as $delegatedAccess) {
```
Comment (ottaviano) [inline src/Entity/Adherent.php:2465]: could we use `$this->getReceivedDelegatedAccessOfType('candidate')` and avoid `'candidate' === $delegatedAccess->getType()` check?

```php
foreach ($this->getReceivedDelegatedAccessOfType('candidate') as $delegatedAccess)
```

File: src/Entity/Adherent.php [hunk 2]
```diff
@@ -2453,6 +2455,50 @@ public function isCandidate(): bool
         return $this->candidateManagedArea instanceof CandidateManagedArea;
     }
 
+    public function isDelegatedCandidate(): bool
+    {
+        return \count($this->getReceivedDelegatedAccessOfType('candidate')) > 0;
```
Comment (ottaviano) [inline src/Entity/Adherent.php:2460]: could you use existing const `TYPE_CANDIDATE`?

File: tests/Controller/EnMarche/CommitteeControllerTest.php [hunk 3]
```diff
@@ -109,7 +110,8 @@ public function testAuthenticatedCommitteeHostCanUnfollowCommittee()
 
         $this->assertResponseStatusCode(Response::HTTP_OK, $this->client->getResponse());
 
-        $unfollowButton = $crawler->filter('.committee-unfollow');
+        $unfollowButton = $crawler->filter(
+            '.committee-unfollow');
```
Comment (ottaviano) [inline tests/Controller/EnMarche/CommitteeControllerTest.php:114]: not really need, no?

File: src/Controller/EnMarche/ManagedUsers/CandidateManagedUsersController.php [hunk 4]
```diff
@@ -40,9 +40,9 @@ protected function createFilterModel(Request $request): ManagedUsersFilter
         return new ManagedUsersFilter(
             SubscriptionTypeEnum::CANDIDATE_EMAIL,
             [$this->getMainUser($session)->getCandidateManagedArea()->getZone()],
-            [],
-            [],
-            [],
+            $this->getRestrictedCommittees($session),
+            $this->getRestrictedCities($session),
+            [$this->getMainUser($session)->getCandidateManagedArea()->getZone()]
```
Comment (ottaviano) [inline src/Controller/EnMarche/ManagedUsers/CandidateManagedUsersController.php:45]: I think we could keep `[]` as value here?

**Review by ottaviano — APPROVED** [review]

**Pull request merged**